src/bosh_azure_cpi/lib/cloud/azure/restapi/vmss_rest.rb

```diff
@@ -0,0 +1,365 @@
+        os_profile['adminUsername'] = vm_params[:ssh_username]
+        os_profile['linuxConfiguration'] = {
+          'disablePasswordAuthentication' => 'true',
+          'ssh' => {
+            'publicKeys' => [
+              {
+                'path'    => "/home/#{vm_params[:ssh_username]}/.ssh/authorized_keys",
+                'keyData' => vm_params[:ssh_cert_data]
+              }
+            ]
+          }
+        }
+      when 'windows'
+        # TODO: add support for the windows.
+        raise ArgumentError, "Unsupported os type: #{vm_params[:os_type]}"
+      else
+        raise ArgumentError, "Unsupported os type: #{vm_params[:os_type]}"
+      end
+
+      network_profile = {
+        'networkInterfaceConfigurations' => []
+      }
+      network_interfaces.each_with_index do |network, index|
+        nic = {
+          'name' => "nic-#{index}",
+          'properties' => {
+            'primary' => index.zero?,
+            'enableIPForwarding' => true,
+            'ipConfigurations' => [
+              {
+                'name' => "ipcfg-#{index}",
+                'properties' => {
+                  'subnet' => {
+                    'id' => network[:subnet][:id]
+                  },
+                  'loadBalancerInboundNatPools' => [],
+                  'loadBalancerBackendAddressPools' => []
+                }
+              }
+            ]
+          }
+        }
+        unless vm_params[:load_balancer].nil?
+          nic['properties']['ipConfigurations'][0]['properties']['loadBalancerBackendAddressPools'].push(
+            'id' => vm_params[:load_balancer][:backend_address_pools][0][:id]
+          )
+        end
+
+        unless network[:network_security_group].nil?
+          nic['properties']['networkSecurityGroup'] = {
+            'id' => network[:network_security_group][:id]
+          }
+        end
+        network_profile['networkInterfaceConfigurations'].push(nic)
+      end
+
+      os_disk = {
+        'createOption' => 'FromImage',
+        'caching'      => vm_params[:os_disk][:disk_caching],
+        # Comment this out, use the default in the image,
+        # will handle this when resolving https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/issues/533
+        # 'managedDisk' => {
+        #   'storageAccountType' => 'Standard_LRS'
+        # }
+      }
+      os_disk['diskSizeGB'] = vm_params[:os_disk][:disk_size] unless vm_params[:os_disk][:disk_size].nil?
+      data_disks = []
+      unless vm_params[:ephemeral_disk].nil?
+        data_disks.push(
+          'createOption' => 'Empty',
+          'diskSizeGB'   => vm_params[:ephemeral_disk][:disk_size],
+          'lun'          => EPHEMERAL_DISK_LUN
+        )
+      end
+      storage_profile = {
+        'imageReference' => {
+          'id' => vm_params[:image_id]
+        },
+        'osDisk' => os_disk,
+        'dataDisks' => data_disks
+      }
+
+      vmss = {
+        'sku' => {
+          'tier' => 'Standard',
```

Comment:
Why not use "Basic"?

Reply:
I think standard is better thing, want to just use the better thing.